Don't generate clients for example internal APIs

thockin opened the pull request:
Remove the generated code and the code that would generate them.  This kills most of the deltas between generate-groups.sh and generate-internal-groups.sh.  That will be cleaned up further later.

/sig architecture
/kind cleanup

```release-note
NONE
```

liggitt commented on staging/src/k8s.io/code-generator/generate-internal-groups.sh:
```diff
@@ -21,14 +21,13 @@ set -o pipefail
-if [ "$#" -lt 5 ] || [ "${1}" == "--help" ]; then
+if [ "$#" -lt 4 ] || [ "${1}" == "--help" ]; then
   cat <<EOF
-Usage: $(basename "$0") <generators> <output-package> <internal-apis-package> <extensiona-apis-package> <groups-versions> ...
```
is this script still actually doing anything with the internal groups? is the intent to delete this script completely by the time you finish carving up https://github.com/kubernetes/kubernetes/pull/114831?

internal and external packages are not always nested (see k8s.io/api and k8s.io/kubernetes/pkg/apis split), so I'm not sure this is still usable by anyone with that split if we collapse the two args to one

thockin replied:
This script will go away completely, and possibly be relaced by something nicer when go workspaces lands.  AFAICT we have no such users in-tree, but I don't know if we want to maintain this script as an external-facing API.  It *only* works with GOPATH.

liggitt replied:
Dropping the script completely and replacing with something nicer sounds great... as a data point, there's a *lot* of invocations of this [in the wild](https://github.com/search?type=code&q=%2F%5C%2Fgenerate-internal-groups.sh+%5B%22a-z%5C%5C%24%5D%2F+-org%3Akubernetes+-org%3Akubernetes-sigs).

I'd like @sttts' feedback on dropping the ability to run this with internal/external API packages located in different places. If we decide that's ok to drop, should we keep the two args, check they are identical, and error if they aren't? That would preserve existing invocations for folks who aren't relying on internal/external API packages being in different locations.

liggitt commented on staging/src/k8s.io/code-generator/generate-internal-groups.sh:
```diff
@@ -57,78 +55,60 @@ shift 5
 if [ "${GENS}" = "all" ] || grep -qw "deepcopy" <<<"${GENS}"; then
   echo "Generating deepcopy funcs"
   "${GOPATH}/bin/deepcopy-gen" \
-      --input-dirs "$(codegen::join , "${ALL_FQ_APIS[@]}")" -O zz_generated.deepcopy \
```
deep-copy generation is still needed on internal types, so it's not clear how the internal package is getting added to the FQ_APIS list

defaulting, conversion, clients, listers, informers are only needed on external types

liggitt replied:
ok, should we stop passing `deepcopy` to generate-internal-groups.sh everywhere then?

thockin replied:
The script, as it exists, will RE-DO the deepcopy  generation.  I wanted this PR to be just a small self-contained part of the larger PR - maybe easier if I just change the client-gen part and leave the rest for the subsequent commits?

liggitt commented:
deletion of the internal clientsets / listers lgtm. a couple questions on the generate-internal-groups.sh change

cici37 commented:
/triage accepted

liggitt commented on staging/src/k8s.io/code-generator/generate-internal-groups.sh:
```diff
@@ -57,78 +55,60 @@ shift 5
-  if [ -n "${INT_APIS_PKG}" ]; then
-    IFS=" " read -r -a APIS <<< "$(printf '%s/ ' "${INT_FQ_APIS[@]}")"
-    "${GOPATH}/bin/client-gen" \
-        --clientset-name "${CLIENTSET_NAME_INTERNAL:-internalversion}" \
```
should we stop setting CLIENTSET_NAME_INTERNAL in all callers?

thockin replied:
yes, thanks.

thockin committed:
Don't generate clients for example internal APIs

Remove the generated code and the code that would generate them.

More cleanups will follow.

thockin commented:
Simplified this diff, leaving the rest for later, thanks.

liggitt commented:
/lgtm
/approve

liggitt commented:
> /usr/bin/env: 'python': No such file or directory

uh...

liggitt commented:
looks like maybe https://github.com/kubernetes/test-infra/pull/28466 broke everything? :)

thockin commented:
There was a PR yesterday to activate python3 - I justrebased this branch to be sure it's included

liggitt commented:
commented at https://github.com/kubernetes/test-infra/pull/28466#issuecomment-1399016336 ... the scripts still relying on python2 live in test-infra and are baked into test images

liggitt commented:
/lgtm
/approve

once presubmits get fixed, should be good to go
